ocs/ocs_twisted.py

```diff
@@ -1,29 +1,113 @@
 import threading
 from contextlib import contextmanager
 
+
 class TimeoutLock:
-    def __init__(self):
+    def __init__(self, default_timeout=0):
+        """
+        Locking mechanism to be used by OCS Agents.
+
+        Args:
+            default_timeout (float, optional):
+                Sets the default timeout value for acquire calls.
+                Defaults to 0.
+        """
         self.job = None
-        self._lock = threading.Lock()
+        self._active = threading.Lock()
+        self._next = threading.Lock()
+        self._default_timeout = default_timeout
+
+    def acquire(self, timeout=None, job=None):
+        """
+        Acquires main lock.
+
+        Args:
+            timeout (float, optional):
+                Sets the timeout for lock acquisition.
+
+                If set to 0 the acquire calls will be non-blocking and will
+                immediately return the result.
 
-    def acquire(self, blocking=True, timeout=-1):
-        return self._lock.acquire(blocking, timeout)
+                If set to any value greater than 0, this will block for that
+                many seconds before returning the result.
```

Comment:
It will return as soon as it has acquired the lock, or within that many seconds if it is unable to acquire the lock.